sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherClientBuilder.java

```diff
@@ -137,8 +142,11 @@ public EventGridPublisherClientBuilder() {
                 context.getHttpRequest().getHeaders().set(AEG_SAS_TOKEN, sasToken.getSignature());
                 return next.process();
             });
-        } else {
+        } else if (keyCredential != null) {
             httpPipelinePolicies.add(new AzureKeyCredentialPolicy(AEG_SAS_KEY, keyCredential));
+        } else {
+            httpPipelinePolicies.add(new BearerTokenAuthenticationPolicy(this.tokenCredential,
+                DEFAULT_EVENTGRID_SCOPE));
```

Comment:
If more than one type of credential is set, we should throw `IllegalStateException` rather than using some priority order (which the user will not know).

Reply:
Updated to throw `IllegalStateException`